lua/go/comment.lua

```diff
@@ -55,30 +55,251 @@ comment.gen = function()
+    if pattern_str then
+      regexes[category] = vim.regex(pattern_str)
+    end
+  end
+
+  -- Define the highlight groups for each category
+  local highlight_groups = {
+    types = 'GoCommentType',
+    functions = 'GoCommentFunction',
+    variables = 'GoCommentVariable',
+    constants = 'GoCommentConstant',
+    parameters = 'GoCommentParameter',
+    -- keywords = 'GoCommentKeyword',
+  }
+
+  if _GO_NVIM_CFG.comment.highlight_groups then
+    for k, v in pairs(_GO_NVIM_CFG.comment.highlight_groups) do
+      highlight_groups[k] = v
+    end
+  end
+
+  -- Define a query to find comment nodes
+  local comment_query = vim.treesitter.query.parse('go', '(comment) @comment')
+
+  -- Iterate over the captures in the query
+  for _, match, _ in comment_query:iter_matches(root, bufnr, 0, -1) do
+    local node = match[1] -- The capture is at index 1
+    if node:type() == 'comment' then
+      local start_row, start_col, end_row, end_col = node:range()
+      for row = start_row, end_row do
+        local line = vim.api.nvim_buf_get_lines(bufnr, row, row + 1, false)[1]
+        local col_start = 0
+        local col_end = #line
+
+        if row == start_row then
+          col_start = start_col
+        end
+        if row == end_row then
+          col_end = end_col
+        end
+
+        -- For each category, search for matches and apply highlights
+        for category, regex in pairs(regexes) do
+          local s = col_start
+          while s < col_end do
+            local m_start, m_end = regex:match_line(bufnr, row, s, col_end)
+            if not m_start then
+              break
+            end
+
+            -- m_start and m_end are relative to 's'
+            local hl_start_col = m_start + s
+            local hl_end_col = m_end + s
+
+            -- Apply the highlight to the matched word
+            vim.api.nvim_buf_add_highlight(
+              bufnr,
+              ns_id,
+              highlight_groups[category],
+              row,
+              hl_start_col,
+              hl_end_col
+            )
+
+            s = hl_end_col -- Move to the end of the current match
+          end
+        end
+      end
+    end
+  end
+end
+
+local function toggle_go_comment_highlight()
+  _GO_NVIM_CFG.comment.enable_highlight = not _GO_NVIM_CFG.comment.enable_highlight
+  if _GO_NVIM_CFG.comment.enable_highlight then
+    highlight_go_code_in_comments()
+  else
+    vim.api.nvim_buf_clear_namespace(0, ns_id, 0, -1)
+  end
+end
+
+comment.highlight = highlight_go_code_in_comments
+comment.toggle_highlight = toggle_go_comment_highlight
+
+vim.api.nvim_create_user_command('ToggleGoCommentHighlight', function()
+  require('go.comment').toggle_highlight()
+end, {})
+
+vim.api.nvim_create_autocmd({ 'BufEnter', 'BufWritePost', 'TextChanged', 'InsertLeave' }, {
+  pattern = { '*.go' },
+  callback = function()
+    if _GO_NVIM_CFG.comment.enable_highlight then
+      require('go.comment').highlight()
```

Comment:
You might want to debounce the function if is run on `TextChanged` and `InsertLeave`.

```lua
-- Debounced function call
local function debounced_function()
  if debounce_timer then
    debounce_timer:stop()
  end
  debounce_timer = vim.defer_fn(function()
    original_function()
  end, 200) -- 200ms debounce time
end

```

Reply:
Thanks, added deboucer